File: js/2279FullBags.mjs
/**
 * @param {number[]} capacity
 * @param {number[]} rocks
 * @param {number} additionalRocks
 * @return {number}
 */
var maximumBags = function (capacity, rocks, additionalRocks) {
  const remaining = capacity.map((c, i) => c - rocks[i]).sort((a, b) => a - b)
  let count = 0
  for (let i = 0; i < remaining.length; i++) {
    if (remaining[i] === 0) {
      count++
    } else {
      if (additionalRocks >= remaining[i]) {
        additionalRocks -= remaining[i]
        count++
      } else {
        break
      }
    }
  }
  return count
}

File: c#/2279FullBags.cs
public class Solution {
    public int MaximumBags(int[] capacity, int[] rocks, int additionalRocks) {
        var deltas = capacity.Zip(rocks,(c,r)=>c-r).OrderBy(e=>e);
        int result = 0;
        foreach (int delta in deltas){
            if (additionalRocks < delta)
                break;
            result+=1;
            additionalRocks-=delta;
        }
        return result;
    }
}